addons/mail/models/mail_template.py

```diff
@@ -104,6 +110,28 @@ def _compute_can_write(self):
         for template in self:
             template.can_write = template in writable_templates
 
+    def _search_can_write(self, operator, value):
+        if operator not in ('=', '!=') or not isinstance(value, bool):
+            raise NotImplementedError('Unsupported search operator')
+        query = Query(self.env, self._table, self._table_sql)
+        self._apply_ir_rules(query, 'write')
+        operator = 'in' if ((operator == '=' and value) or (operator == '!=' and not value)) else 'not in'
+        return [('id', operator, query)]
+
+    @api.depends_context('uid')
+    def _compute_can_unlink(self):
+        unlinkable_templates = self._filter_access_rules('unlink')
+        for template in self:
+            template.can_unlink = template in unlinkable_templates
+
+    def _search_can_unlink(self, operator, value):
+        if operator not in ('=', '!=') or not isinstance(value, bool):
+            raise NotImplementedError('Unsupported search operator')
+        query = Query(self.env, self._table, self._table_sql)
+        self._apply_ir_rules(query, 'unlink')
+        operator = 'in' if ((operator == '=' and value) or (operator == '!=' and not value)) else 'not in'
+        return [('id', operator, query)]
```

Comment:
All that is useless, you can just fetch all template current can read, then call filter_access_rules_python on it normally. You won't have millions of templates so that's ok to fetch them all then filter, instead of adding compute / search fields on model, which quite bloat it without much usage.